From 0c5c10b3c5cb460a978ff685a99cb6b1be4a84b9 Mon Sep 17 00:00:00 2001
From: Vabhav Sharma <vabhav.sharma@nxp.com>
Date: Tue, 26 Mar 2019 19:43:35 +0530
Subject: [PATCH 677/706] serial: fsl_lpuart: enable LE support

Enable reading lpuart device-tree node to check if IP with
compatible fsl,ls1021a-lpuart is LE and update endiannes
type of uart node in driver.
By default driver is used in BE mode,

Signed-off-by: Vabhav Sharma <vabhav.sharma@nxp.com>
(cherry picked from commit b4d6c533839808dbfeedcded28a222e08acf0b09)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/tty/serial/fsl_lpuart.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/serial/fsl_lpuart.c b/drivers/tty/serial/fsl_lpuart.c
index 50b6746a8b5d..cdf15ca7901a 100644
--- a/drivers/tty/serial/fsl_lpuart.c
+++ b/drivers/tty/serial/fsl_lpuart.c
@@ -3,6 +3,8 @@
  *  Freescale lpuart serial port driver
  *
  *  Copyright 2012-2014 Freescale Semiconductor, Inc.
+ *  Copyright 2019 NXP
+ *
  */
 
 #if defined(CONFIG_SERIAL_FSL_LPUART_CONSOLE) && defined(CONFIG_MAGIC_SYSRQ)
@@ -2170,7 +2172,10 @@ static int lpuart_probe(struct platform_device *pdev)
 		return ret;
 	}
 	sport->port.irq = ret;
-	sport->port.iotype = sdata->iotype;
+	if (of_property_read_bool(pdev->dev.of_node, "little-endian"))
+		sport->port.iotype = UPIO_MEM32;
+	else
+		sport->port.iotype = sdata->iotype;
 	if (lpuart_is_32(sport))
 		sport->port.ops = &lpuart32_pops;
 	else
-- 
2.17.1

